policy/compressor.go

```diff
@@ -0,0 +1,132 @@
+package policy
+
+import (
+	"sync"
+
+	"github.com/m3db/m3cluster/kv"
+	"github.com/m3db/m3metrics/generated/proto/schema"
+	"github.com/m3db/m3x/instrument"
+	"github.com/m3db/m3x/log"
+)
+
+// Compressor maintains a mapping of policies to integers. It is useful when encoding policies
+// so that an integer representation of a policy can be sent instead of the entire Policy
+type Compressor interface {
+	Get(p Policy) (int64, bool)
+}
+
+type compressorMap map[Policy]int64
+
+type noopCompressor struct{}
+
+// NewNoopCompressor returns a new Compressor which never matches a policy
+func NewNoopCompressor() Compressor {
+	return noopCompressor{}
+}
+
+func (m noopCompressor) Get(p Policy) (int64, bool) {
+	return -1, false
```

Comment:
Actually, if we define a constant for invalid id's its a little redundant to return a boolean as well. Personally, I'm more in favor of return the bool since it's more idiomatic with other go data structures, but I don't feel strongly either way.